pinot-controller/src/main/java/org/apache/pinot/controller/recommender/rules/RulesToExecute.java

```diff
@@ -160,9 +169,14 @@ public boolean isRecommendRealtimeProvisioning() {
     return _recommendRealtimeProvisioning;
   }
 
+  public boolean isRecommendSegmentSize() {
+    return _recommendSegmentSize;
+  }
+
   // Be careful with the sequence, each rule can execute individually
   // but a rule may depend on its previous rule when they both fired
   public enum Rule {
+    SegmentSizeRule, // This rule must be the first rule. It provides segment count, segment size, numRows in segments which are used in other rules. It also adjust cardinality per segment for different columns.
```

Comment:
I think we should always run this rule since computing CPS (used in all subsequent rules) and numSegments (partition rule) are important. 

So the user can still disable it via RulesToExecute as they can do for other rules. But internally we always run to compute the information needed by other rules. If disabled, we just don't return the recommendations (S, NRPS and S_size)

Reply:
The only adjustment to cardinality is that we cap it to the segment size. Per previous comments, it's not useful if we run this rule for realtime-only tables.
For offline or hybrid table, this rule needs to be run first. So I'm going to modify recommender execution to enforce these to conditions:
- disable the rule for realtime-only tables
- enable the rule for offline or hybrid tables